src/gui/MainWindow.cpp

```diff
@@ -570,13 +570,21 @@ void MainWindow::addSpacingToToolBar( int _size )
 								7, _size );
 }
 
+
+
+
 SubWindow* MainWindow::addWindowedWidget(QWidget *w, Qt::WindowFlags windowFlags)
 {
 	// wrap the widget in our own *custom* window that patches some errors in QMdiSubWindow
 	auto win = new SubWindow(m_workspace->viewport(), windowFlags);
 	win->setAttribute(Qt::WA_DeleteOnClose);
 	win->setWidget(w);
-	if (w && w->sizeHint().isValid()) {win->resize(w->sizeHint());}
+	if (w && w->sizeHint().isValid()) {
+		std::size_t titleBarHeight = win->titleBarHeight();
+		std::size_t frameWidth = win->frameWidth();
```

Comment:
You're using `std::size_t` here, but the return value is `int`. `QSize` takes `int` as constructor parameters. Should probably just make these `auto`.

Reply:
No, `int` is one letter shorter. :grin:

Reply:
Bad reason but I'm not going to argue.

Reply:
Sorry, I think your argument makes sense. If Qt should ever switch from int to long, we are now prepared with `auto`.